CONTRIBUTING.md

```diff
@@ -0,0 +1,47 @@
+# Contributing to Irving
+This document is intended to make contributing to this project easier, more transparent, and more standardized.
+
+## Commitizen
+This project uses [Commitizen](https://github.com/commitizen/cz-cli) and [Conventional Changelog](https://github.com/conventional-changelog/conventional-changelog) to standardize commit messages and changelogs. This means when you enter `git commit` to commit your code you will see a (maybe) unfamiliar, interactive series of questions. These questions will result in a standardized format for your commit message. If you have any questions about the possible categories, scopes, etc. please create an issue in GitHub and we will do our best to respond.
+
+Note: For merge commits, just use `chore` for the type of change, `merge` for the scope, and `merge commit` for the message. Likely, however, it will not matter what you put into Commitizen as you'll just get a normal merge commit message such as `Merge branch 'my-branch' into canary`.
+
+## Git Workflow and Branches
+Specific branches will be used for specific purposes in this repo. Considering this, the lifecycle of a new feature, bugfix, etc. should largely follow the steps below.
+
+### If you are a community contributor:
+1. Fork the Irving repo.
+2. `git checkout master` - check out the NPM package release branch.
+3. `git checkout -b feature/issue-17-branch-title` - create a new feature branch based on master. If your branch relates to a specific GitHub issue, reference that issue in your branch name (and all subsequent commits).
+4. `npm run test` and/or `npm run test:watch` - If you've fixed a bug or added a new feature, please write tests! These commands can be used to run tests from within the package you are modifying or at the root of the repo if you want to run all tests for all packages.
+5. Work on your branch, using `npm link` where necessary to test out your code.
+6. When ready, push your code and create a pull request into the `master` branch. Your code will be reviewed by someone at Alley.
+7. Once your code passes review, someone at Alley will merge it into `master`.
+8. Your code should be available on npm in the next prerelease in the npm `@beta` tag.
+9. Keep an eye on the releases page. When you see your code has been released, _install and test it!_.
+10. Assuming neither you nor someone at Alley finds issues with your code, it will also be included in the next `@rc` and, subsequently, `@latest` releases.
+
+### If you are a part of the Alley organization and have publish capabilities for the npm packages:
```

Comment:
I think breaking the process for publishing releases out to a separate .md file might help to keep this out of the way of people who are only trying to contribute. Otherwise, most of the rest of the process should be similar between contributors opening PRs from a fork or people who are pushing branches directly to the repo (i.e. Alley folks)

Reply:
That's a very good point, I'll submit something for that.